kornia/losses/focal.py

```diff
@@ -122,3 +122,93 @@ def forward(  # type: ignore
             input: torch.Tensor,
             target: torch.Tensor) -> torch.Tensor:
         return focal_loss(input, target, self.alpha, self.gamma, self.reduction, self.eps)
+
+
+def binary_focal_loss_with_logits(
+        inp: torch.Tensor,
+        target: torch.Tensor,
+        alpha: float = .25,
+        gamma: float = 2.0,
+        reduction: str = 'none',
+        eps: float = 1e-8) -> torch.Tensor:
+    r"""Function that computes Binary Focal loss.
+
+    """
+    if not isinstance(inp, torch.Tensor):
+        raise TypeError("Input type is not a torch.Tensor. Got {}"
+                        .format(type(inp)))
+
+    if not len(inp.shape) >= 2:
+        raise ValueError("Invalid input shape, we expect BxCx*. Got: {}"
+                         .format(inp.shape))
+
+    if inp.size(0) != target.size(0):
+        raise ValueError('Expected input batch_size ({}) to match target batch_size ({}).'
+                         .format(inp.size(0), target.size(0)))
+
+    if not inp.device == target.device:
+        raise ValueError(
+            "input and target must be in the same device. Got: {} and {}".format(
+                inp.device, target.device))
+
+    probs = torch.sigmoid(inp)
+    target = target.unsqueeze(dim=1)
+    loss_tmp = -alpha * torch.pow((1. - probs), gamma) * target * torch.log(probs + eps) \
+               - (1 - alpha) * torch.pow(probs, gamma) * (1. - target) * torch.log(1. - probs + eps)
+    loss_tmp = loss_tmp.squeeze(dim=1)
+
+    if reduction == 'none':
+        loss = loss_tmp
+    elif reduction == 'mean':
+        loss = torch.mean(loss_tmp)
+    elif reduction == 'sum':
+        loss = torch.sum(loss_tmp)
+    else:
+        raise NotImplementedError("Invalid reduction mode: {}"
+                                  .format(reduction))
+    return loss
+
+
+class BinaryFocalLossWithLogits(nn.Module):
+    r"""Criterion that computes Focal loss.
+
+    According to [1], the Focal loss is computed as follows:
```

Comment:
I think the docs error comes from the reference here. We support bibtex style for the bibliography. Check this:
https://github.com/kornia/kornia/blob/master/kornia/feature/hardnet.py#L17
https://github.com/kornia/kornia/blob/master/docs/source/references.bib